opensearch_py_ml/ml_commons/ml_commons_client.py

```diff
@@ -606,3 +609,91 @@ def delete_task(self, task_id: str) -> object:
             method="DELETE",
             url=API_URL,
         )
+
+    def _get_profile(self, payload: Optional[dict] = None):
+        """
+        Get the profile using the given payload.
+
+        :param payload: The payload to be used for getting the profile. Defaults to None.
+        :type payload: Optional[dict]
+        :return: The response from the server after performing the request.
+        :rtype: Any
+        """
+        validate_profile_input(None, payload)
+        return self.client.transport.perform_request(
+            method="GET", url=f"{ML_BASE_URI}/{self.API_ENDPOINT}", body=payload
+        )
+
+    def _get_models_profile(
+        self, model_id: Optional[str] = "", payload: Optional[dict] = None
+    ):
+        """
+        Get the profile of a model.
+
+        Args:
+            model_id (str, optional): The ID of the model. Defaults to "".
+            payload (dict, optional): Additional payload for the request. Defaults to None.
+
+        Returns:
+            dict: The response from the API.
+        """
+        self._validate_input(model_id, payload)
+
+        url = f"{ML_BASE_URI}/{self.API_ENDPOINT}/models/{model_id if model_id else ''}"
+        return self.client.transport.perform_request(
+            method="GET", url=url, body=payload
+        )
+
+    def _get_tasks_profile(
+        self, task_id: Optional[str] = "", payload: Optional[dict] = None
+    ):
+        """
+        Retrieves the profile of a task from the API.
+
+        Parameters:
+            task_id (str, optional): The ID of the task to retrieve the profile for. Defaults to an empty string.
+            payload (dict, optional): Additional payload for the request. Defaults to None.
+
+        Returns:
+            dict: The profile of the task.
+
+        Raises:
+            ValueError: If the input validation fails.
+
+        """
+        self._validate_input(task_id, payload)
+
+        url = f"{ML_BASE_URI}/{self.API_ENDPOINT}/tasks/{task_id if task_id else ''}"
+        return self.client.transport.perform_request(
+            method="GET", url=url, body=payload
+        )
+
+    def get_profile(self, profile_type: str = "all", ids: Optional[List[str]] = None, request_body: Optional[dict] = None) -> dict:
```

Comment:
Let's try to add doc like this way: https://github.com/opensearch-project/opensearch-py-ml/blob/main/opensearch_py_ml/ml_commons/ml_commons_client.py#L70-L101


This will help to create documentation automatically. 

Otherwise, overall looks good to me.